lib/node-utils.ts

```diff
@@ -181,3 +181,48 @@ export function isRenderVariableDeclarator(
 
   return false;
 }
+
+type GetTestingLibraryUtilImportArgs = {
+  context: any,
+  utilName: string,
+  extraModules?: string[],
+};
+
+export function getTestingLibraryUtilImport(options: GetTestingLibraryUtilImportArgs): null | string {
+
+  /* Here we need to:
+   - look for imports of shape: import { <utilName> } from '@testing-library/whatever';
+   - look for imports of shape: import { <utilName> as <alias> } from '@testing-library/whatever';
+   - look for imports of shape: import * as testingLibrary from '@testing-library/whatever';
+   - look for imports of shape: const render = require('@testing-library/whatever').render;
+
+   From 3rd case, this method can't return what's the imported name for `utilName` as it's contained within an object.
+   What should we do in that case? Return an object or array to indicate if `utilName` has been imported directly or
+   under a namespace?
+  */
```

Comment:
Here you can see the first problem. With the current implementation I had, I found this:

Working fine if name imported from testing library
```javascript
// node
import { screen } from '@testing-library/framework';
// ...

// rule implementation
const importedScreen = getTestingLibraryUtilImport({ context, utilName: 'screen' });
console.log(importedScreen) // 'screen'
```

Working fine if rename imported from testing library
```javascript
// node
import { screen as testingScreen } from '@testing-library/framework';
// ...

// rule implementation
const importedScreen = getTestingLibraryUtilImport({ context, utilName: 'screen' });
console.log(importedScreen) // 'testingScreen'
```

But I don't know what to do here
```javascript
// node
import * as testingLibrary from '@testing-library/framework';
// ...

// rule implementation
const importedScreen = getTestingLibraryUtilImport({ context, utilName: 'screen' });
console.log(importedScreen) // null
```

For the 3rd case, this method must return something else to indicate `'screen'` has been imported, but its namespaced under `testingLibrary` object rather than imported directly. How should the method return that then?

Reply:
🤔 
what about returning an object instead? in addition to the `alias` (that might be the object exported, an alias, or a full object name that contains all the exported stuff), a enum/flag indicating the type of export so then you know how to treat the string?

Reply:
Yeah, I think this is probably the best option. It'll return an object including the imported name and the context of the import. I need to think what to use for the context of the import. I guess I can use same types from AST, so from the example before it would return:

```javascript
// node
import { screen as testingScreen } from '@testing-library/framework';
// ...

// rule implementation
const importedScreen = getTestingLibraryUtilImport({ context, utilName: 'screen' });
console.log(importedScreen) // { Identifier: 'testingScreen', type: 'ImportSpecifier' }
```

```javascript
// node
import * as testingLibrary from '@testing-library/framework';
// ...

// rule implementation
const importedScreen = getTestingLibraryUtilImport({ context, utilName: 'screen' });
console.log(importedScreen) // { Identifier: 'testingLibrary', type: 'ImportNamespaceSpecifier' }
```

Thoughts?